canvas/gradient.go

```diff
@@ -68,6 +68,9 @@ func (g *LinearGradient) Hide() {
 
 // Move the gradient to a new position, relative to its parent / canvas
 func (g *LinearGradient) Move(pos fyne.Position) {
+	if g.Position() == pos {
+		return
+	}
 	g.baseObject.Move(pos)
 
 	repaint(g)
```

Comment:
See the comment below, I think it would be nice to be consistent and apply the same approach here and everywhere else.